### PR TITLE
Many the setup and deploy changes!

### PR DESCRIPTION
**Pull request by abortz**
Major items:
- Factored out virtualenv stuff -- let developers use one or not, their choice. We use one in production.
- Integrated z3 into repo: submodule and makefile integration (awkward because I didn't want to overlay changes onto the z3 repo)
- (More) mature deployment setup: user-mode upstart service management, and git hook-based deploys. (systemd looks like its actually easier to get user services working, but not going to try on 14.04LTS)
- Still memory leaks, but I've capped the size and upstart will helpfully keep the service up.

Take a look @eseidel !

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,51 +6,53 @@ scripts_dir = scripts
...
+	@echo "Run 'make deps' to bring your dependencies up-to-date."
 	@echo "Run 'make check' to check against the current baseline and 'make accept' to set a new baseline from the last results."
 
-sayc-env:
-	@virtualenv sayc-env
+deps: z3py_install
+	@pip install --upgrade $(PYTHON_PACKAGES)
+
+$(z3_dir)/build/config.mk:
+	@cd $(z3_dir) && python scripts/mk_make.py --prefix=$(python_prefix)
```
Comment (eseidel) [inline Makefile]: z3 has to be system-installed I assume, and not virtual-env?

**Comment by eseidel** [comment]
lgtm

**Comment by eseidel** [comment]
i should clearly setup travis for saycbridge. :)

File: Makefile (hunk 1 above)
Comment (abortz) [inline Makefile]: No the Makefile will do an in-tree build of z3, lib and py components, and drop them in the right place. Well, the version I have locally will ;-) It needed a small tweak.

**Comment by abortz** [comment]
Ha I was just looking at that! Concerned that travis-ci gets to see my private Dropbox repos -- silly Github permissions/user model...

**Pull request merged**